pypeit/pypmsgs.py

```diff
@@ -135,11 +137,16 @@ def _initialize_log_file(self, log=None):
         """
         Expects self._log is already None.
         """
-        if log is None:
+
+        if log is not None:
+            if isinstance(log, io.IOBase):
+                self._log = log
+            else:
+                # Initialize the log
+                self._log = open(log, 'w')
+        else:
```

Comment:
I think the code is cleaner when minimizing nesting.  So I would instead:
```
if log is None:
    return

self._log = log if isinstance(log, io.IOBase) else open(log, 'w')
```
but that's just my preference.  No need to change if you don't agree.

Reply:
I think you're right. The trinary operator in C/C++ has made me too averse for using this construct in python